tests/test_middleware.py

```diff
@@ -14,84 +22,103 @@ def base64_encode_for_py2or3(text):
-        request = self.create_request()
-        response = self.middleware.process_request(request)
+class TestFeinCMSLoginRequiredMiddleware(RequestTestCase):
+    middleware = FeinCMSLoginRequiredMiddleware(check=False)
+    AUTH_STATE = AccessState.STATE_AUTH_ONLY
+    OTHER_STATE = ('other', 'Other state')
 
-        self.assertEqual(response, None)
+    class DummyFeinCMSPage:
+        def __init__(self, access_state):
+            self.access_state = access_state
 
-    @mock.patch(PROTECTED_URLS_PATH, PROTECTED_URLS)
-    def test_skip_middleware_if_url_is_not_protected(self):
-        """Assert 'url' is not protected when it is not in LOGIN_PROTECTED_URLS."""
-        request = self.create_request()
-        response = self.middleware.process_request(request)
-        self.assertEqual(response, None)
+    def make_request(self, auth, access_state, **kwargs):
+        """Create a request with a suitable feincms_page attribute."""
+        request = self.create_request(auth=auth, **kwargs)
+        request.feincms_page = self.DummyFeinCMSPage(access_state=access_state)
+        return request
 
-    def test_skip_middleware_if_user_is_authenticated(self):
-        """Assert middleware does not redirect if user is authenticated."""
-        request = self.create_request(user=AuthenticatedUser())
+    def test_unprotected_url(self):
+        request = self.make_request(
+            auth=False,
+            access_state=AccessState.STATE_ALL_ALLOWED
+        )
         response = self.middleware.process_request(request)
-        self.assertEqual(response, None)
+        self.assertIsNone(response)
 
-    def test_403_result_if_non_get_request(self):
-        """Assert other methods than GET are not allowed when user is not logged in."""
-        request = self.create_request(method='POST')
+    def test_other_state_url(self):
+        request = self.make_request(auth=False, access_state=self.OTHER_STATE)
         response = self.middleware.process_request(request)
-        self.assertEqual(response.status_code, 403)
+        self.assertIsNone(response)
 
-    def test_redirect_if_all_is_well(self):
-        """Assert user is redirected to the login page when urls are protected."""
-        request = self.create_request()
+    def test_is_auth(self):
+        request = self.make_request(auth=True, access_state=self.AUTH_STATE)
         response = self.middleware.process_request(request)
-        self.assertEqual(response.status_code, 302)
+        self.assertIsNone(response)
 
-    @override_settings(LOGIN_URL='/login/')
-    def test_login_url(self):
-        """Assert redirect accepts LOGIN_URL as string."""
-        request = self.create_request()
+    def test_non_auth_get(self):
+        request = self.make_request(auth=False, access_state=self.AUTH_STATE)
         response = self.middleware.process_request(request)
-        self.assert_redirect_url(response, '/login/?next=url/')
```

Comment:
@adam-thomas I know this was a while ago but do you have any recollection of this? I would have thought it would be pretty straight forward to restore this functionality but I am curious as to whether there was a reason it was removed.